test/Microsoft.AspNetCore.Http.Tests/Internal/DefaultHttpRequestTests.cs

```diff
@@ -194,6 +195,54 @@ public void Cookies_GetAndSet()
             Assert.Equal(new[] { "name2=value2" }, cookieHeaders);
         }
 
+        [Fact]
+        public void RouteValues_GetAndSet()
+        {
+            var context = new DefaultHttpContext();
+            var routeValuesFeature = context.Features.Get<IRouteValuesFeature>();
+            // No feature set for initial DefaultHttpRequest
+            Assert.Null(routeValuesFeature);
+
+            var request = context.Request;
+            var routeValues = request.RouteValues;
+            // Route values defaults to null
+            Assert.Null(routeValues);
```

Comment:
Yeah, this shouldn't be null. The caller shouldn't have to check for null before looking up a value in the collection.